File: zhilian-backend/src/main/java/com/zhilian/zhilianbackend/entity/Manufacture.java
package com.zhilian.zhilianbackend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

/**
 * @Author: 6017
 * @Date: 2026/3/9 21:44
 * @Param: 
 * @Return: 
 * @Description: 制造企业表实体类，对应manufacture表
**/
@Getter
@Setter
@Accessors(chain = true)
@TableName("manufacture")
public class Manufacture implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 企业唯一标识
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 关联user.id，不可为空，确保企业有归属
     */
    @TableField("user_id")
    private Long userId;

    /**
     * 企业全称
     */
    @TableField("company_name")
    private String companyName;

    /**
     * 所在区域（深圳/东莞/惠州/广州等）
     */
    @TableField("region")
    private String region;

    /**
     * 详细地址
     */
    @TableField("address")
    private String address;

    /**
     * 联系人
     */
    @TableField("contact_person")
    private String contactPerson;

    /**
     * 联系人电话
     */
    @TableField("contact_phone")
    private String contactPhone;

    /**
     * 规模：微型、小型、中型、大型
     */
    @TableField("scale")
    private String scale;

    /**
     * 员工人数
     */
    @TableField("employee_count")
    private Integer employeeCount;

    /**
     * 年营收（万元）
     */
    @TableField("annual_revenue")
    private BigDecimal annualRevenue;

    /**
     * 主营产品类型（如PCB、半导体、消费电子）
     */
    @TableField("product_type")
    private String productType;

    /**
     * 企业简介
     */
    @TableField("description")
    private String description;

    /**
     * Logo图片URL
     */
    @TableField("logo")
    private String logo;

    /**
     * 成立日期
     */
    @TableField("established_date")
    private Date establishedDate;

    /**
     * 审核状态：待审核、通过、驳回
     */
    @TableField("audit_status")
    private String auditStatus;

    /**
     * 审核意见（驳回时填写）
     */
    @TableField("audit_remark")
    private String auditRemark;

    /**
     * 审核时间
     */
    @TableField("audit_time")
    private Date auditTime;

    /**
     * 审核人ID，关联user.id
     */
    @TableField("audit_user_id")
    private Long auditUserId;

    /**
     * 逻辑删除时间，''1970-01-01 00:00:00'' 表示未删除，其他时间表示已删除
     */
    @TableField("deleted")
    @TableLogic
    private Date deleted;

    /**
     * 记录创建时间
     */
    @TableField("create_time")
    private Date createTime;

    /**
     * 最后更新时间
     */
    @TableField("update_time")
    private Date updateTime;
}

File: zhilian-frontend/src/api/manufacture.js
import request from "@/utils/request";

/**
 * 获取制造企业列表（公共列表）
 * @param {Object} params 查询参数
 * @param {number} [params.page=1] 页码
 * @param {number} [params.size=10] 每页条数
 * @param {string} [params.region] 区域筛选
 * @param {string} [params.scale] 规模筛选：micro/small/medium/large
 * @param {string} [params.productType] 主营产品类型（模糊匹配）
 * @returns {Promise<{
 *   total: number,
 *   records: Array<{
 *     id: number,
 *     companyName: string,
 *     region: string,
 *     scale: string,
 *     productType: string,
 *     contactPerson: string,
 *     contactPhone: string,
 *     auditStatus: string
 *   }>
 * }>} 分页结果（仅包含审核通过的企业）
 */
export function getManufactureList(params) {
  return request({
    url: "/manufacture/list",
    method: "get",
    params,
  });
}

/**
 * 获取制造企业详情
 * @param {number} id 企业ID
 * @returns {Promise<{
 *   id: number,
 *   userId: number,
 *   companyName: string,
 *   region: string,
 *   address: string,
 *   contactPerson: string,
 *   contactPhone: string,
 *   scale: string,
 *   employeeCount: number,
 *   annualRevenue: number,
 *   productType: string,
 *   description: string,
 *   logo: string,
 *   establishedDate: string,
 *   auditStatus: string,
 *   auditRemark: string | null,
 *   auditTime: string | null,
 *   createTime: string,
 *   updateTime: string
 * }>} 企业详细信息
 */
export function getManufactureDetail(id) {
  return request({
    url: `/manufacture/${id}`,
    method: "get",
  });
}

/**
 * 新增制造企业
 * @param {Object} data 企业信息
 * @param {string} data.companyName 企业全称
 * @param {string} [data.region] 区域
 * @param {string} [data.address] 详细地址
 * @param {string} [data.contactPerson] 联系人
 * @param {string} [data.contactPhone] 联系电话
 * @param {string} [data.scale] 规模枚举
 * @param {number} [data.employeeCount] 员工人数
 * @param {number} [data.annualRevenue] 年营收（万元）
 * @param {string} [data.productType] 主营产品类型
 * @param {string} [data.description] 企业简介
 * @param {string} [data.logo] Logo图片URL
 * @param {string} [data.establishedDate] 成立日期
 * @returns {Promise<{ id: number, auditStatus: string }>} 新增成功返回ID和审核状态（pending）
 */
export function addManufacture(data) {
  return request({
    url: "/manufacture",
    method: "post",
    data,
  });
}

/**
 * 修改制造企业信息
 * @param {number} id 企业ID
 * @param {Object} data 要修改的字段（全部可选）
 * @returns {Promise<null>} 无返回数据
 */
export function updateManufacture(id, data) {
  return request({
    url: `/manufacture/${id}`,
    method: "put",
    data,
  });
}

/**
 * 删除制造企业（逻辑删除）
 * @param {number} id 企业ID
 * @returns {Promise<null>} 无返回数据
 */
export function deleteManufacture(id) {
  return request({
    url: `/manufacture/${id}`,
    method: "delete",
  });
}
